### ISSUE TITLE
A11y - Search box close butting inaccessible using keyboard

### ISSUE THREAD

**Issue by ericabr** [carbon-design-system/carbon-components-react#830]
## A11y for search box
A11y manual check fails keyboard navigation: Using the search box in our UI doesn't pass a11y tests because the icon used to clear our the contents can't be reached via the keyboard.

## Detailed description
A person using the keyboard cannot use the "x" to clear the contents of the box. The user will have to leave focus of the item, return the focus and delete the contents manually.
![focus_search_close_btn](https://user-images.githubusercontent.com/33731164/39159516-f08086ee-4733-11e8-9733-5df2ab319ca1.gif)

## Steps to reproduce the issue
This can be reproduced in the storybook

**Comment by joshblack** [comment]
Hi @ericabr! 👋 Thanks so much for taking the time to make this issue! Definitely is an a11y violation, we should probably be coding the clear icon up as a button to help remedy this 🙂 Will gladly accept any PRs that help fix this!

**Comment by joshblack** [comment]
Created an issue on `carbon-components` to fix this upstream, once that's resolved we can tackle this one in the React repo 😄

**Comment by joshblack** [comment]
Depends on carbon-design-system/carbon-components#830

**Comment by joshblack** [comment]
Upstream PR just got merged, just need to update the markup in the component 👍
